example/src/libretro.rs

```diff
@@ -65,37 +65,35 @@ fn key_to_retro_button(key: keyboard::Key) -> RetroJoypadButton {
 }
 
 impl RetroCore for LibretroCore {
+  type SpecialGameType = ();
+  type SubsystemMemoryType = ();
```

Comment:
I really wish associated type defaults were stable :(

Reply:
I'm not 100% happy with this part of the design but I'm not sure if the alternatives are worth it.

I could remove the need for these types in the common case by hiding `load_game_special` and its associated env calls behind a feature flag. That "solves" the problem insofar as the average user that doesn't need subsystems isn't forced to declare types they won't use. But adding or removing parts of a trait with a feature flag seems unconventional?

Another possibility is to separate the subsystem functionality into an additional trait and add a new pattern to the macro (e.g. `libretro_core!(Emulator as SubsystemCore)`. Moving the subsystem functionality into another trait seems more conventional, but I'm not sure about having to invoke the macro another way. Seems like it could be a source of trouble.

The other option is to ditch the associated types, wrap the id in a newtype and make it the user's problem to convert that into a useful enum.

Reply:
I don't have enough exposure to this part of the libretro API to give any guidance here. I don't plan on releasing a `0.2` any time soon, so we can iterate on this in the interim.